Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -45,32 +39,91 @@ extension Command {
         switch conf {
         case .Debug:
-            args += ["-g", "-O0"]
+            return ["-g", "-O0"]
         case .Release:
-            args += ["-O2"]
+            return ["-O2"]
+        }
+    }
+}
+
+private extension Sources {
+    func compilePathsForBuildDir(wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
+        return relativePaths.map { source in
+            let path = Path.join(root, source)
+            let object = Path.join(wd, "\(source).o")
+            let deps = Path.join(wd, "\(source).d")
+            return (source, path, object, deps)
         }
+    }
+}
+
+extension Command {
+    static func compile(clangModule module: ClangModule, configuration conf: Configuration, prefix: String) -> ([Command], Command) {
+
+        let wd = Path.join(prefix, "\(module.c99name).build")
+        let mkdir = Command.createDirectory(wd)
+
+        ///------------------------------ Compile -----------------------------------------
+        var compileCommands = [Command]()
+        let dependencies = module.dependencies.map{ $0.targetName }
+        let basicArgs = module.basicArgs + module.includeFlags + module.optimizationFlags(conf)
+        for path in module.sources.compilePathsForBuildDir(wd) {
+            var args = basicArgs
+            args += ["-MMD", "-MT", "dependencies", "-MF", path.deps]
+            args += ["-c", path.source, "-o", path.object]
+
+            let node = "<\(module.name).\(path.filename)>"
+
+            let clang = ClangTool(desc: "Compiling \(module.name) \(path.filename)",
+                                  inputs: dependencies + [path.source, mkdir.node],
+                                  outputs: [path.object, node],
+                                  args: args,
+                                  deps: path.deps)
+
+            let command = Command(node: node, tool: clang)
+
+            compileCommands.append(command)
+        }
+
+
+        ///FIXME: This probably doesn't belong here
+        ///------------------------------ Product -----------------------------------------
+
+        var args = module.basicArgs
+        args += module.optimizationFlags(conf)
+        args += ["-L\(prefix)"]
+        args += module.linkFlags
+        args += module.sources.compilePathsForBuildDir(wd).map{$0.object}
 
-        args += module.sources.paths
-        
         if module.type == .Library {
             args += ["-shared"]
         }
-        
-        args += ["-o", productPath]
 
-        let clang = ShellTool(
-            description: "Compiling \(module.name)",
-            inputs: inputs,
-            outputs: [productPath, module.targetName],
-            args: [Toolchain.clang] + args)
+        let productPath = Path.join(prefix, module.type == .Library ? "lib\(module.c99name).so" : module.c99name)
+        args += ["-o", productPath]
 
+        let clang = ClangTool(desc: "Linking \(module.name)",
```

Comment:
This should probably just use a shell tool, the clang tool was really intended specifically for compilation, not when clang is being used as a driver for the linker.